scripts/gui/CourseDisplay.lua

```diff
@@ -259,22 +259,36 @@ function SimpleCourseDisplay:updateChangesBetween(firstIx, secondIx)
 end
 
 --- Changes the visibility of the course.
-function SimpleCourseDisplay:updateVisibility(visible, onlyStartStopVisible)
+function SimpleCourseDisplay:updateVisibility(visibilityData)
 	if self.course then
 		local numWp = self.course:getNumberOfWaypoints()
 		for j = 1, numWp do
-			if self.signs[j] then 
+			if self.signs[j] then
+				local visible = false
+				local onlyStartStopVisible = false
+				if visibilityData.visiblityMode == CpVehicleSettings.SHOW_COURSE_ALL then
+					visible = visibilityData.isControlled ~= nil and visibilityData.isControlled or visibilityData.isControlled == nil

```

Comment:
just `visible = visibilityData.isControlled` would work fine, no need to check for nil if we use it as boolean.

But again, this logic should be in the caller, as it was before.